fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -342,6 +342,7 @@ private void init(String s, Type type) throws AnalysisException {
             Preconditions.checkArgument(type.isDateType());
             TemporalAccessor dateTime = null;
             boolean parsed = false;
+            s = s.replace("'", "");
```

Comment:
In the function `partitionInfo.toPartitionDesc`，it will call `partitionInfo.toPartitionValue`. In this function，there is a `expr.toSql()` to construct partitionValue. `expr.toSql()` will contain "'". However, it is not permitted when executing the sql with symbol "'"